python/ray/ml/config.py

```diff
@@ -116,6 +119,149 @@ def as_placement_group_factory(self) -> "PlacementGroupFactory":
         return PlacementGroupFactory(bundles, strategy=self.placement_strategy)
 
 
+@dataclass
+@PublicAPI(stability="alpha")
+class DatasetConfig:
+    """Configuration for ingest of a single Dataset.
+
+    These configs define how the Dataset should be read into the DataParallelTrainer.
+    It configures the preprocessing, splitting, and ingest strategy per-dataset.
+
+    DataParallelTrainers declare default DatasetConfigs for each dataset passed in the
+    ``datasets`` argument. Users have the opportunity to selectively override these
+    configs by passing the ``dataset_config`` argument. Trainers can also define user
+    customizable values (e.g., XGBoostTrainer doesn't support streaming ingest).
+    """
+
+    # TODO(ekl) could we unify DataParallelTrainer and Trainer so the same data ingest
+    # strategy applies to all Trainers?
+
+    # Whether to fit preprocessors on this dataset. This can be set on at most one
+    # dataset at a time.
+    # True by default for the "train" dataset only.
+    fit: Optional[bool] = None
+
+    # Whether the dataset should be split across multiple workers.
+    # True by default for the "train" dataset only.
+    split: Optional[bool] = None
+
+    # Whether to raise an error if the Dataset isn't provided by the user.
+    # True by default for the "train" dataset only.
```

Comment:
I'm wondering if we should not make any Datasets required. This would be to support piecemeal usage of the Trainers without needing to use Ray Datasets.

Reply:
How about making it required=False by default in `DataParallelTrainer`? Subclasses like HF trainer can make it required though if needed.

Reply:
Yep that sg!